charmhelpers/contrib/network/files/check_sriov_numvfs.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+"""
+Check to verify the number of virtual functions (VFs) active on network
+interfaces. Based on /sys/class/net/<interface>/device/sriov_numvfs and
+/sys/class/net/<interface>/device/sriov_totalvfs.
+
+    usage: check_sriov_numvfs.py [-h] sriov_numvfs [sriov_numvfs ...]
+
+    Check sriov numvfs configuration
+
+    positional arguments:
+    sriov_numvfs  format: <interface>:<numvfs>
+
+For each interfaces:numvfs pair given it verifies that:
+    1. VFs are enabled
+    2. VFs are configured are the same as specified in the check parameter
+    3. VFs configured do not exceed the maximum available on the interface
+
+Example: ./check_sriov_numvfs.py ens3f0:32 ens3f1:32 ens6f0:32 ens6f1:32
+"""
+
+import argparse
+import os.path
+import sys
+
+
+DEVICE_TEMPLATE = "/sys/class/net/{0}"
+SRIOV_NUMVFS_TEMPLATE = "/sys/class/net/{0}/device/sriov_numvfs"
+SRIOV_TOTALVFS_TEMPLATE = "/sys/class/net/{0}/device/sriov_totalvfs"
```

Comment:
Hmm - why are we checking this one?  This is the total number of VFs that the PF supports?  Customers could have valid reasons for configuring less VFs than the number the PF supports, e.g. the less VFs you have, the more channels/queues you have per VF.

Reply:
This is the number of VFs that the PF supports. 
We are checking this to make sure we are not trying to configure more VFs than the PF supports. This could happen if we have different NICs in a group of servers with different capabilities.
Currently, if the number of VFs set on the PF is higher than totalvfs, it just sets it to totalvfs (silent fail). 

If `numvfs > totalvfs` then `numvfs != sriov_numvfs` will also fail, however the help text will report a more accurate reason and therefore hopefully help to resolve it faster.

Reply:
Can you elaborate a bit more on why this is a critical alert when the requested VFs cannot be provided by the card? It seems conceivable to me that 2 cards with different capabilities across different units have the same interface name but one can serve 128 VFs and another 64. Setting the value to 128 maxes out both available VFs but it is fewer than 128*2.